sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -82,8 +148,8 @@ case class StaticInvoke(
     }
 
     val code = s"""
-      ${argGen.map(_.code).mkString("\n")}
-      $setIsNull
+      $argCode
+      boolean ${ev.isNull} = $resultIsNull;
       final $javaType ${ev.value} = ${ev.isNull} ? ${ctx.defaultValue(dataType)} : $callFunc;
```

Comment:
The `${ev.isNull}` at line 153 can be `$resultIsNull`. So the optimization for `false ? ${ctx.defaultValue(dataType)} : $callFunc;` can work here too?